client/src/utils/api.ts

```diff
@@ -0,0 +1,278 @@
+import downloadZip from "./downloadZip";
+
+const BASE_URL = 'http://localhost:8080';
+interface CustomResponse {
+	zip?: string;
+	error?: boolean;
+	errorData?: string;
+	errorTitle?: string;
+	message?: string;
+}
+type StatusProps = {
+	message: string;
+	isSuccessful: boolean;
+	isEnd: boolean;
+};
+
+const settings = {
+	notificationLoaded: '.loaded',
+	notificationFailed: '.loaded-fail',
+	statusName: {
+		functionCreated: 'Function Created',
+		zipCreated: 'Zip Created',
+		zipUploaded: 'Zip Uploaded',
+		lambdaCreated: 'Lambda Created',
+		rangeTimeAdded: 'Range time added',
+	},
+	endPointUrls: {
+		modifyFileUrl: '/api/modify-file',
+		createZipUrl: '/api/create-zip',
+		uploadZipUrl: '/api/uploadZip',
+		createLambdaUrl: '/api/create-lambda',
+		addEventBridgeUrl: '/api/add-eventbridge',
+		modifyFileLocalUrl: '/api/modify-file-local',
+		createCfZip: '/api/create-cf-zip',
+	},
+};
+export type FetcherOptions = {
+	queryString: string
+	variables?: FetcherVariables
+}
+
+export type FetcherVariables = { [key: string]: string | any | undefined }
+
+
+
+class Api {
```

Comment:
The API class has multiple responsibilities. It has to be split to a logic responsible for API call itself(header, response code etc.) and method like initPage or downloadCFTemplate has to be separate services(classes or function)